codelab-master/apps/playground/src/app/quiz/shared/services/quiz.service.ts

```diff
@@ -0,0 +1,217 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { BehaviorSubject } from 'rxjs';
+import { Howl } from 'howler';
+
+import { QUIZ_DATA } from '@codelab-quiz/shared/quiz-data';
+import { Option, Quiz, QuizQuestion } from '@codelab-quiz/shared/models/';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class QuizService {
+  quizData: Quiz[] = JSON.parse(JSON.stringify(QUIZ_DATA));
+  question: QuizQuestion;
+  questions: QuizQuestion[];
+  answers: number[];
+  multipleAnswer: boolean;
+  totalQuestions: number;
+  currentQuestionIndex = 1;
+
+  quizId: string;
+  startedQuizId: string;
+  continueQuizId: string;
+  completedQuizId: string;
+  indexOfQuizId: number;
+  quizCompleted: boolean;
+  status: string;
+
+  correctAnswers = [];
+  correctAnswersForEachQuestion = [];
+  correctAnswerOptions: number[] = [];
+  numberOfCorrectAnswers: number;
+  correctAnswersCountSubject = new BehaviorSubject<number>(0);
+
+  userAnswers = [];
+  previousUserAnswers: any[] = [];
+  previousUserAnswersText: any[] = [];
+  previousUserAnswersInnerText = [];
+  previousUserAnswersTextSingleAnswer: string[] = [];
+  previousUserAnswersTextMultipleAnswer: string[] = [];
+
+  explanation: string;
+  explanationText: string;
+  correctOptions: string;
+  correctMessage: string;
+
+  isAnswered: boolean;
+  checkedShuffle: boolean;
+
+  correctSound = new Howl({
```

Comment:
ahaha, there are sounds 🔥🔥🔥

Reply:
Yes, the incorrect answer sound is a bit annoying :)